TST: make_archive_device cannot make a Lightpath device that passes LightpathMixin checks

## Expected Behavior
We should be able to make an archived device for all the relevant devices in `all_pcdsdevice_classes`

## Current Behavior
Calling `make_archived_device(LightpathMixin)` fails because `lightpath_cpts` has not been defined, and `LightpathMixin.__init_subclass__` checks for this.

Note that the `__init_subclass__` behavior is working as intended: subclasses of these LightpathMixin's are being checked for lightpath-compatibility.  

In the below example, `LightpathInOutCptMixin` would be created fine since it is marked as a mixin, while `ArchiverLightpathInOutCptMixin` fails, since it is a subclass of `LIghtpathInOutCptMixin` without `lightpath_cpts` defined

## Possible Solution
- add a condition to `all_pcdsdevice_classes` that skips the LightpathMixin classes, which used to get skipped because they didn't inherit from `ophyd.Device`? (hacky but will work)
- Change the init_subclass behavior in the mixins?  (not my favorite option)

## Steps to Reproduce (for bugs)
1. Run the test suite with an updated pcdsdevices (pcds-5.5.0)

<details> 
    <summary> pytest traceback </summary>

    ```
    >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> traceback >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
    
    cls = <class 'pcdsdevices.interface.LightpathInOutCptMixin'>
    
        @all_pcdsdevice_classes
        def test_switch_control_layer(cls):
            lp_mixin = getattr(cls, '_lightpath_mixin', '')
            is_mixin = getattr(cls, '_is_mixin', '')
            if lp_mixin != '':
                print(cls.__name__, is_mixin, lp_mixin)
    >       make_archived_device(cls)
    
    atef/tests/test_archived_device.py:26: 
    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
    atef/archive_device.py:544: in make_archived_device
        return switch_control_layer(
    atef/archive_device.py:643: in switch_control_layer
        new_class = type(
    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
    
    cls = <class 'atef.archive_device.ArchiverLightpathInOutCptMixin'>, kwargs = {}
    
        def __init_subclass__(cls, **kwargs):
            print(cls.__name__, cls._lightpath_mixin)
            super().__init_subclass__(**kwargs)
            if cls._lightpath_mixin:
                # Child of cls will inherit this as False
                cls._lightpath_mixin = False
            else:
                if not cls.lightpath_cpts:
    >               raise NotImplementedError(
                        'Did not implement LightpathMixin properly.  '
                        'Must supply a list of components (lightpath_cpts)'
                    )
    E               NotImplementedError: Did not implement LightpathMixin properly.  Must supply a list of components (lightpath_cpts)
    
    ../pcdsdevices/pcdsdevices/interface.py:1726: NotImplementedError
    >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> entering PDB >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
    ```

</details>


## Context
I'm not entirely sure where this issue goes. The bug is in atef, but the fix might more naturally go in pcdsdevices. 

## Your Environment
pcds-5.5.0

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -17,7 +17,7 @@ repos:
         exclude: '^(conda-recipe/meta.yaml)$'
     -   id: debug-statements
 
--   repo: https://gitlab.com/pycqa/flake8.git
+-   repo: https://github.com/pycqa/flake8.git
     rev: 3.9.0
     hooks:
     -   id: flake8

--- a/atef/tests/test_archived_device.py
+++ b/atef/tests/test_archived_device.py
@@ -8,11 +8,15 @@ import pytest
 from ..archive_device import ArchivedValue, make_archived_device
 from . import conftest
 
+# skip these devices, subclassing is too involved to be automated
+SKIP_CLS = ['LightpathMixin', 'LightpathInOutCptMixin']
+
 all_pcdsdevice_classes = pytest.mark.parametrize(
     "cls",
     [
         pytest.param(cls, id=f"{cls.__module__}.{cls.__name__}")
-        for cls in pcdsdevices.tests.conftest.find_all_device_classes()
+        for cls
+        in pcdsdevices.tests.conftest.find_all_device_classes(skip=SKIP_CLS)
     ]
 )
 

